### PR TITLE
Add Quantumdex Bid Adapter

### PR DESCRIPTION
**Pull request by thuyhq**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Add new bid adapter for Quantumdex

- Test parameters for validating bids
Banner Test Parameters
```
var adUnits = [
  {
    code: 'test-div',
    mediaTypes: {
      banner: {
        sizes: [[300, 250], [300,600]]
      }
    },
    bids: [
      {
          bidder: 'quantumdex',
          params: {
              siteId: 'quantumdex-site-id',
          }
      }
    ]
  }
];
```
Video Test Parameters
```
var videoAdUnit = {
  code: 'test-div',
  sizes: [[640, 480]],
  mediaTypes: {
    video: {
      playerSize: [[640, 480]],
      context: 'instream'
    },
  },
  bids: [
    {
      bidder: 'quantumdex',
      params: {
        siteId: 'quantumdex-site-id',
      }
    }
  ]
};
```

- contact email of the adapter’s maintainer: ken@quantumdex.io
- [x] official adapter submission
- PR link on the docs repo: https://github.com/prebid/prebid.github.io/pull/1951

### CODE REVIEW

**Comment by thuyhq** [comment]
Hello maintainer team,
My pull request getting error with circleci build.
But this error not come from my commit.
error
Can you help me check it?

**Comment by Fawke** [comment]
> Hello maintainer team,
> My pull request getting error with circleci build.
> But this error not come from my commit.
> error
> Can you help me check it?
@thuyhq 
It's failing because of some race condition, it's not related to your PR, don't worry about it. I see that it's passing for your second commit.

File: modules/quantumdexBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            type: px.type,
+            url: px.url
+          });
+        });
+      }
+    } catch (e) { }
+    return syncs;
+  },
+
+  onTimeout: function (timeoutData) {
```
Comment (Fawke) [inline modules/quantumdexBidAdapter.js:154]: not required to declare an empty function.

**Comment by thuyhq** [comment]
> Hi @thuyhq,
> 
> Thanks for the PR. While reviewing it, I've made few observations which you might wanna address:
> 
> 1. Actual test parameter not provided in the md file. We need to see the response coming back from the server and see the ad rendered on the screen. It'll be great if you can provide an actual value for `siteId`, instead of a placeholder value.
> 2. I see you support GDPR. If your bidder is enrolled for TCF v2.0 and has a `gvl id`, you might wanna include that as a new property in your spec. You can refer to appnexusBidAdapter.js for reference.
> 3. I see you are using certain properties of the `window` object. What if the Ad is rendered inside and iframe? Won't you need to do, `window.top` instead?

Hello @Fawke ,
Thank you for your review.

1. Our endpoint provides a demo siteId "quantumdex-site-id". It always returns ads, you can use it to check.
2. We are currently not registered for TCF v2.0. There will be specific plans for future implementation.
3. I have fixed it.

Thank you very much.

**Comment by thuyhq** [comment]
> Hi @thuyhq,
> 
> Thanks for making the changes. I tested with the test parameter and was able to get a banner response from your endpoint but I see an error when testing `instream`.
> 
> Here's the error.
> Screenshot 2020-05-04 at 1 58 46 PM
> 
> Am guessing some issue with the creative vast. Not a big deal, but just wanted to bring that to your notice.
> 
> Also, I've a suggestion. In places where you are using `window.top`, you might wanna look at the `bidRequest.referrerInfo` object. It has certain properties and methods which might be helpful in your case. Here is the doc for it: http://prebid.org/dev-docs/bidder-adaptor.html. Please search for `refererInfo` once you visit that page.

Hi @Fawke ,

Thanks for your great feedback.
I updated my code using the referrerInfo object as you suggested.
Please help me check back.

Regarding the instream ad error issue you encountered. I think it is not an error from the quantumdex adapter.
- Our endpoint has returned vastUrl and vastXml demo. I successfully broadcast an ad when testing with Google's Video Suite Inspector.
- I tried with bidder appnexus test parameters and got the same error you encountered.
This is probably the fault of Google DFP Adunit that you used to test. This error occurs when I test with "/19968336/prebid_cache_video_adunit"

**Pull request merged**